biz.aQute.launchpad/src/aQute/launchpad/Launchpad.java

```diff
@@ -870,6 +870,37 @@ public Framework getFramework() {
 			.start();
 	}
 
+	/**
+	 * Runs the given code within the context of a synthetic bundle. Creates a
+	 * synthetic bundle and adds the supplied class to it using
+	 * {@link BundleBuilder#addResourceWithCopy}. It then loads the class using
+	 * the synthetic bundles class loader, and instantiates it using the default
+	 * constructor.
+	 * 
+	 * @param clazz the class to instantiate within the context of the
+	 *            framework.
+	 * @return The instantiated object.
+	 * @throws NoSuchMethodException if the supplied class doesn't have a
+	 *             default constructor.
+	 * @see BundleBuilder#addResourceWithCopy(Class)
+	 */
+	public <T> T instantiateInFramework(Class<? extends T> clazz) {
+		try {
+			clazz.getConstructor();
+		} catch (NoSuchMethodException e) {
+			Exceptions.duck(e);
+		}
+		Bundle b = bundle().addResourceWithCopy(clazz)
+			.start();
+		try {
+			@SuppressWarnings("unchecked")
+			Class<? extends T> insideClass = (Class<? extends T>) b.loadClass(clazz.getName());
+			return insideClass.newInstance();
```

Comment:
This should be `return insideClass.getConstructor().newInstance();`

Reply:
Can you explain your reasoning a bit? Is it that you would prefer instantiateInFramework() to throw an InvocationTargetException rather than throwing the raw exception?

Reply:
See https://github.com/bndtools/bnd/commit/8ae6af3bb8d252463d9f6e5cbca46044e15251b4

Reply:
Thanks for educating me! Also ironic that the reason `Class.newInstance()` has been deprecated is the same reason that I originally preferred it over `getConstructor()`, and that throughout Launchpad we have gone to efforts to bypass checked exceptions...

I'm wondering if perhaps `insideClass.getDeclaredConstructor().newInstance()` might not be better (as `getConstructor()` will only work with public constructors). Your thoughts?

Reply:
It is best to limit to the default (public) constructor. Then people can control whether they want the constructor called. And we avoid needing setAccessible and all that baggage. The OSGi specs (e.g.) only use public constructors.

Reply:
I mostly agree, but one pushback for consideration: Restricting it to public definitely makes sense for production code; however this is code that is typically used for testing, and often in testing you want to do things and exercise code paths that you would usually avoid in a production environment. Hence the thinking that it might be nice to make it easier to access them.

Reply:
Perhaps, but it seems a small chance that there is a non-public no-param constructor (that is meaningful) and not a public no-param constructor. The more interesting case is that there are constructors which take arguments and that is not useful here since this code only deals with no-param constructor

Reply:
Fair enough. Adding support for constructors with args was something I considered, and I think it would be a worthwhile addition, but as my immediate need was only for a no-arg constructor I tried to limit the scope to correspond to my limited time. :smile: